packages/clerk-js/src/core/resources/Base.ts

```diff
@@ -129,6 +130,14 @@ export abstract class BaseResource {
     method = 'POST',
     path,
   }: BaseMutateParams): Promise<this> {
+    if (typeof body === 'object') {
```

Comment:
This shouldn't be a concern of the fapi client - lets discuss alternatives

Reply:
Do you have anything in mind. This is the only place to intercept I think and it might not be a concert of the fapi client but it removes complexity from a lot of places. We can always refactor if we can't find an alternative now.

Reply:
I prefer some duplication instead of making fapiClient aware of URL-based session syncing. These are two completely different concerns and should be handled accordingly. 

`actionCompleteRedirectUrl` can only be found in SignIn and SignUp resources. If not all, most `redirectUrl` props pass through SignIn/ SignUp as well - can we check that this is the case indeed and refactor accordingly? 

Another util we could explore is `buildSSOCallbackURL`.

Reply:
I'll check if there's a way to abstract it without the current implementation, but the point was to make this something we don't need to worry about anymore. I'll get back to you.

Reply:
I made some changes. PTAL.